libc/CMakeLists.txt

```diff
@@ -39,7 +39,11 @@ set(LIBC_BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR})
 set(LIBC_ENABLE_USE_BY_CLANG OFF CACHE BOOL "Whether or not to place libc in a build directory findable by a just built clang")
 
 # Defining a global namespace to enclose all libc functions.
-set(LIBC_NAMESPACE "__llvm_libc_${LLVM_VERSION_MAJOR}_${LLVM_VERSION_MINOR}_${LLVM_VERSION_PATCH}_${LLVM_VERSION_SUFFIX}"
+set(default_namespace "__llvm_libc")
```

Comment:
is this change related to the rest of this PR? It seems like an odd inclusion

Reply:
Why would `LLVM_VERSION_MAJOR` not be set? Does this happen for the runtimes build? (if so, cc @petrhosek )

Reply:
This happens if you build it directly out of the `runtimes/` directory. We specifically build everything else independent from LLVM if possible so I don't think this is a surprising change to have a fallback if the user is building out-of-tree or manually.

Reply:
I wonder if building via the runtimes directory SHOULD set `LLVM_VERSION_MAJOR`? Otherwise libcxx, compiler-rt, and now llvmlibc wont be able to key off this value.

The non `LIBC_NAMESPACE` changes LGTM, if you want to split this PR in two, I'd accept the second half.  For the first, I suspect we may want to fix the runtimes CMake itself.

Reply:
We already want to support out-of-tree builds, I don't see the issue with setting a sane default for those cases.

Reply:
> We already want to support out-of-tree builds

What do you mean? Folks still need to use llvm/ or runtimes/ if they use cmake. If they're not using cmake, then setting a default namespace for cmake is irrelevant.

Reply:
All this does is change `__llvm_libc_____` into `__llvm_libc` is this a blocking issue or can I merge the patch as-is.